### PR TITLE
hypre: add in hipblas dependency due to superlu-dist

### PR DESCRIPTION
**Pull request by balay**

### CODE REVIEW

**Review by eugeneswalker — APPROVED** [review]

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -258,6 +259,8 @@ def configure_args(self):
 
         if "+rocm" in spec:
             rocm_pkgs = ["rocsparse", "rocthrust", "rocprim", "rocrand"]
+            if "+superlu-dist" in spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hypre/package.py:262]: FYI. We're now encouraging use of `self.spec.satisfies()` to check variants of the package.  As the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints state:

`The difference between the two functions is that satisfies() tests whether spec constraints overlap at all, while in tests whether a spec or any of its dependencies satisfy the provided spec.`

**Review by tldahlgren — COMMENTED** [review]
FYI.

File: var/spack/repos/builtin/packages/hypre/package.py (hunk 1 above)
Comment (balay) [reply var/spack/repos/builtin/packages/hypre/package.py:262]: perhaps this can be a global change to hypre spec in a different PR @osborn9 ?

Comment (osborn9) [reply var/spack/repos/builtin/packages/hypre/package.py:262]: Yes - I'll make those updates in a separate PR.

**Review by osborn9 — APPROVED** [review]
Thanks @balay! Looks good.

**Pull request merged**